lab_2_keywords_cooccurrence/main.py

```diff
@@ -3,12 +3,47 @@
 Extract keywords based on co-occurrence frequency
 """
 from pathlib import Path
-from typing import Optional, Sequence, Mapping
+from typing import Any, Optional, Sequence, Mapping
 
 KeyPhrase = tuple[str, ...]
 KeyPhrases = Sequence[KeyPhrase]
 
 
+def check_sequence(obj: Any, element_type: type, can_be_empty: bool) -> bool:
+    """
+    Checks if the object is a list containing elements of a certain type
+
+    :param obj: an object that is expected to be a list
+    :param element_type: the expected type of list elements
+    :param can_be_empty: True if the object can be empty, False otherwise
+    :return: bool - True if the object is list containing elements of the given type, False otherwise
+    """
+    if (can_be_empty is False and obj) or (can_be_empty is True and not obj):
+        if isinstance(obj, list or tuple):
```

Comment:
does it always work?
in my experience, if object was indeed a tuple, such `isinstance` would return False as it does not support `or` operator

Reply:
done